ipn/local.go

```diff
@@ -690,7 +696,15 @@ func (b *LocalBackend) SetPrefs(new *Prefs) {
 	}
 
 	b.updateFilter(b.netMapCache)
-	b.updateDNSMap(b.netMapCache)
+	// TODO(dmytro): when Prefs gain an EnableTailscaleDNS toggle, updateDNSMap here.
+
+	turnDerpOff := new.DisableDERP && !old.DisableDERP
+	turnDerpOn := !new.DisableDERP && old.DisableDERP
+	if turnDerpOff {
+		b.e.SetDERPMap(nil)
+	} else if turnDerpOn {
+		b.e.SetDERPMap(b.netMapCache.DERPMap)
```

Comment:
is `b.netMapCache` guaranteed to be non-nil at this point? Might want to add that condition to `turnDerpOn` on line 702, at least for clarity and robustness (even if it's already guaranteed somehow else).